lib/youtube/youtube.ex

```diff
@@ -7,6 +7,45 @@ defmodule Youtube do
   plug(Tesla.Middleware.BaseUrl, @base_url)
   plug(Youtube.OAuth.Middleware)
 
+  def get_channel_videos(channel_id) do
```

Comment:
I struggled a bit to understand this function and the error handling because it only returns the last video, if it's the only one.
Maybe you could rename it accordingly, or just return the list and let the caller raise if it isn't the expected length? Maybe you can even just do `[the_only_one] = get_channel_videos()` where you use it, and when the pattern fails you get to see the unexpected data.